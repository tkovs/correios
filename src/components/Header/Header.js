import React, { useState } from 'react'
import PropTypes from 'prop-types'
import { StyleSheet, StatusBar } from 'react-native'
import { Appbar, Menu } from 'react-native-paper'

import { colors } from '../../styles/theme'

const Header = ({ enableBackButton, goBack, title, subtitle }) => {
  const [menuVisible, setMenuVisible] = useState(false)

  return (
    <Appbar.Header style={styles.header}>
      {enableBackButton && <Appbar.BackAction onPress={goBack} />}
      <StatusBar backgroundColor={colors.blue} />
      <Appbar.Content title={title} subtitle={subtitle} />
      <Menu
        visible={menuVisible}
        onDismiss={() => setMenuVisible(false)}
        anchor={
          <Appbar.Action
            color="white"
            icon="dots-vertical"
            onPress={() => setMenuVisible(true)}
          />
        }
      >
        <Menu.Item onPress={() => {}} title="Editar" />
        <Menu.Item onPress={() => {}} title="Excluir" />
      </Menu>
    </Appbar.Header>
  )
}

const styles = StyleSheet.create({
  header: {
    backgroundColor: colors.blue,
  },
})

Header.defaultProps = {
  enableBackButton: false,
  goBack: null,
  subtitle: null,
}

Header.propTypes = {
  enableBackButton: PropTypes.bool,
  goBack: PropTypes.func,
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string,
}

export default Header
