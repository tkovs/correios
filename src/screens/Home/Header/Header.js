import React from 'react'
import { StyleSheet, StatusBar } from 'react-native'
import { Appbar } from 'react-native-paper'

import { colors } from '../../../styles/theme'

const Header = () => {
  return (
    <Appbar.Header style={styles.header}>
      <StatusBar backgroundColor={colors.blue} />
      <Appbar.Content title="Correios" subtitle="Rastreio de pacotes" />
    </Appbar.Header>
  )
}

const styles = StyleSheet.create({
  header: {
    backgroundColor: colors.blue,
  },
})

export default Header
