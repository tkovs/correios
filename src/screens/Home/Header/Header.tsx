import React, { FC } from 'react'
import { StyleSheet, StatusBar } from 'react-native'
import { Appbar } from 'react-native-paper'

import { colors } from '../../../styles/theme'

interface HeaderProps {
  updatePackets: () => void
}

type Props = HeaderProps

const Header: FC<Props> = ({ updatePackets }: Props) => (
  <Appbar.Header accessibilityStates={['disabled']} style={styles.header}>
    <StatusBar backgroundColor={colors.blue} />
    <Appbar.Content
      accessibilityStates={['disabled']}
      title="Correios"
      subtitle="Rastreio de pacotes"
    />
    <Appbar.Action
      accessibilityStates={['disabled']}
      icon="refresh"
      onPress={updatePackets}
    />
  </Appbar.Header>
)

const styles = StyleSheet.create({
  header: {
    backgroundColor: colors.blue,
  },
})

export default Header
