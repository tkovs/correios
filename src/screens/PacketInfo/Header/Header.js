import React, { useState } from 'react'
import PropTypes from 'prop-types'
import { StyleSheet, StatusBar } from 'react-native'
import { Appbar, Menu } from 'react-native-paper'

import { colors } from '../../../styles/theme'

const Header = ({ goBack, title }) => {
  const [menuVisible, setMenuVisible] = useState(false)

  return (
    <Appbar.Header style={styles.header}>
      <Appbar.BackAction onPress={goBack} />
      <StatusBar backgroundColor={colors.blue} />
      <Appbar.Content title={title} subtitle="Informações do envio" />
      <Menu
        visible={menuVisible}
        onDismiss={() => setMenuVisible(false)}
        anchor={
          <Appbar.Action
            color="white"
            icon="dots-vertical"
            onPress={() => setMenuVisible(true)}
          />
        }
      >
        <Menu.Item onPress={() => {}} title="Editar" />
        <Menu.Item onPress={() => {}} title="Excluir" />
      </Menu>
    </Appbar.Header>
  )
}

const styles = StyleSheet.create({
  header: {
    backgroundColor: colors.blue,
  },
})

Header.defaultProps = {
  goBack: null,
}

Header.propTypes = {
  goBack: PropTypes.func,
  title: PropTypes.string.isRequired,
}

export default Header
