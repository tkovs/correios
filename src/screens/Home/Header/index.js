import React from 'react'
import { useNavigation } from '@react-navigation/native'

import Header from './Header'

function HeaderContainer(props) {
  const { goBack } = useNavigation()

  return <Header {...props} goBack={goBack} />
}

export default HeaderContainer
