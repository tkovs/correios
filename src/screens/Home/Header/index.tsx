import React, { FC } from 'react'
import { connect, MapDispatchToProps } from 'react-redux'

import Header from './Header'
import { updatePackets as updatePacketsAction } from '../../../store/actions/packets'
import { PacketsActionTypes } from '../../../store/actions/packets/types'

interface DispatchProps {
  updatePackets: () => PacketsActionTypes
}

type Props = DispatchProps

const HeaderContainer: FC<Props> = ({ updatePackets }: Props) => {
  return <Header updatePackets={updatePackets} />
}

const mapDispatchToProps: MapDispatchToProps<DispatchProps, {}> = (
  dispatch
) => ({
  updatePackets: () => dispatch(updatePacketsAction()),
})

export default connect(null, mapDispatchToProps)(HeaderContainer)
