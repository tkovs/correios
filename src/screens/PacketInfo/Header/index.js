import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { useNavigation } from '@react-navigation/native'
import { removePacket, archivePacket } from '../../../store/actions/packets'
import { addFeedback } from '../../../store/actions/feedback'

import Header from './Header'

function HeaderContainer(props) {
  const { archivePacket: archive, removePacket: remove, code, ...rest } = props
  const { goBack } = useNavigation()

  return (
    <Header
      {...rest}
      goBack={goBack}
      removePacket={() => remove(code)}
      archivePacket={() => archive(code)}
    />
  )
}

HeaderContainer.propTypes = {
  removePacket: PropTypes.func.isRequired,
  archivePacket: PropTypes.func.isRequired,
  code: PropTypes.string.isRequired,
}

const mapDispatchToProps = dispatch => ({
  removePacket: code => dispatch(removePacket(code)),
  archivePacket: code => dispatch(archivePacket(code)),
  addFeedback: message => dispatch(addFeedback(message)),
})

export default connect(
  null,
  mapDispatchToProps
)(HeaderContainer)
