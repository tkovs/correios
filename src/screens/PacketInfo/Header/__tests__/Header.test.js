import 'react-native'
import React from 'react'
import { render } from '@testing-library/react-native'

import Header from '../Header'

const mockAddFeedback = jest.fn()
const mockRemovePacket = jest.fn()
const mockArchivePacket = jest.fn()

// TODO: Migrate test to redux

describe('Header', () => {
  it('renders correctly', () => {
    const mockTitle = 'iPad'
    const { baseElement } = render(
      <Header
        addFeedback={mockAddFeedback}
        archivePacket={mockArchivePacket}
        title={mockTitle}
        removePacket={mockRemovePacket}
      />
    )
    expect(baseElement).toMatchSnapshot()
  })
})
