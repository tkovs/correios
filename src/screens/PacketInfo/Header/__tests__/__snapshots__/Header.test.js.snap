// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <SafeAreaView
      style={
        Array [
          Object {
            "backgroundColor": "#01417F",
            "elevation": 4,
            "zIndex": 0,
          },
          Object {
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 3,
              "width": 0,
            },
            "shadowOpacity": 0.24,
            "shadowRadius": 4,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#01417F",
            "elevation": 0,
            "flexDirection": "row",
            "height": 56,
            "marginTop": 0,
            "paddingHorizontal": 4,
          }
        }
      >
        <TouchableHighlight
          accessibilityComponentType="button"
          accessibilityLabel="Back"
          accessibilityRole="button"
          accessibilityStates={Array []}
          accessibilityTraits="button"
          activeOpacity={0.85}
          centered={true}
          delayPressOut={100}
          disabled={true}
          hitSlop={
            Object {
              "bottom": 6,
              "left": 6,
              "right": 6,
              "top": 6,
            }
          }
          onPress={null}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "margin": 6,
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 18,
                  "height": 36,
                  "width": 36,
                },
                undefined,
                undefined,
              ],
            ]
          }
          underlayColor="rgba(255, 255, 255, 0.16)"
        >
          <View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "justifyContent": "center",
                  },
                  Object {
                    "height": 24,
                    "width": 24,
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "transform": Array [
                      Object {
                        "rotate": "0deg",
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "justifyContent": "center",
                      },
                      Object {
                        "height": 24,
                        "transform": Array [
                          Object {
                            "scaleX": 1,
                          },
                        ],
                        "width": 24,
                      },
                    ]
                  }
                >
                  <Image
                    source={
                      Object {
                        "testUri": "../../react-native-paper/lib/commonjs/assets/back-chevron.png",
                      }
                    }
                    style={
                      Array [
                        Object {
                          "height": 21,
                          "resizeMode": "contain",
                          "width": 21,
                        },
                        Object {
                          "tintColor": "#ffffff",
                        },
                      ]
                    }
                  />
                </View>
              </View>
            </View>
          </View>
        </TouchableHighlight>
        <TouchableWithoutFeedback>
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                  "paddingHorizontal": 12,
                },
                Array [
                  Object {
                    "marginLeft": 8,
                  },
                  false,
                  undefined,
                ],
              ]
            }
          >
            <Text
              accessibilityRole="header"
              accessibilityTraits="header"
              accessible={true}
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "color": "#ffffff",
                      "fontFamily": "System",
                      "fontWeight": "400",
                    },
                    Object {
                      "fontSize": 17,
                    },
                    undefined,
                  ],
                ]
              }
            />
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "fontSize": 11,
                    },
                    Object {
                      "color": "rgba(255, 255, 255, 0.7)",
                    },
                    undefined,
                  ],
                ]
              }
            >
              Informações do envio
            </Text>
          </View>
        </TouchableWithoutFeedback>
        <View
          collapsable={false}
        >
          <TouchableHighlight
            accessibilityComponentType="button"
            accessibilityRole="button"
            accessibilityStates={Array []}
            accessibilityTraits="button"
            activeOpacity={0.85}
            centered={true}
            delayPressOut={100}
            disabled={false}
            hitSlop={
              Object {
                "bottom": 6,
                "left": 6,
                "right": 6,
                "top": 6,
              }
            }
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Array [
                  Object {
                    "alignItems": "center",
                    "justifyContent": "center",
                    "margin": 6,
                    "overflow": "hidden",
                  },
                  Object {
                    "borderRadius": 18,
                    "height": 36,
                    "width": 36,
                  },
                  undefined,
                  undefined,
                ],
              ]
            }
            underlayColor="rgba(255, 255, 255, 0.16)"
          >
            <View>
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                    },
                    Object {
                      "height": 24,
                      "width": 24,
                    },
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "opacity": 1,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "transform": Array [
                        Object {
                          "rotate": "0deg",
                        },
                      ],
                    }
                  }
                >
                  <Text
                    accessibilityElementsHidden={true}
                    allowFontScaling={false}
                    importantForAccessibility="no-hide-descendants"
                    pointerEvents="none"
                    style={
                      Array [
                        Object {
                          "color": "white",
                          "fontSize": 24,
                        },
                        Array [
                          Object {
                            "transform": Array [
                              Object {
                                "scaleX": 1,
                              },
                            ],
                          },
                          Object {
                            "backgroundColor": "transparent",
                          },
                        ],
                        Object {
                          "fontFamily": "Material Design Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                  >
                    
                  </Text>
                </View>
              </View>
            </View>
          </TouchableHighlight>
        </View>
      </View>
    </SafeAreaView>
  </View>
  <View
    style={
      Object {
        "bottom": 0,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <View
      pointerEvents="box-none"
      style={
        Object {
          "bottom": "100%",
          "flexDirection": "row",
          "justifyContent": "flex-end",
          "paddingBottom": 4,
          "paddingEnd": 4,
          "position": "absolute",
          "width": "100%",
        }
      }
    >
      <TouchableWithoutFeedback
        hitSlop={
          Object {
            "bottom": 4,
            "left": 4,
            "right": 4,
            "top": 4,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(250, 186, 48, 0.95)",
              },
              Object {
                "borderRadius": 14,
                "height": 28,
                "justifyContent": "center",
                "paddingHorizontal": 12,
              },
            ]
          }
        >
          <Text
            numberOfLines={1}
            style={
              Object {
                "color": "rgba(255, 255, 255, 1)",
                "fontSize": 12,
                "includeFontPadding": false,
                "lineHeight": 16,
              }
            }
          >
            Dismiss All
          </Text>
        </View>
      </TouchableWithoutFeedback>
    </View>
    <ScrollView
      data={
        Array [
          Object {
            "category": "Warning: Failed prop type: The prop \`title\` is marked as required in \`Header\`, but its value is \`undefined\`.
    in Header",
            "warnings": Array [
              YellowBoxWarning {
                "message": Object {
                  "content": "Warning: Failed prop type: The prop \`title\` is marked as required in \`Header\`, but its value is \`undefined\`.
    in Header",
                  "substitutions": Array [],
                },
                "stack": Array [
                  Object {
                    "arguments": Array [],
                    "column": 15,
                    "file": "/Users/tkovs/Code/tracking/node_modules/prop-types/checkPropTypes.js",
                    "lineNumber": 20,
                    "methodName": "printWarning",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 11,
                    "file": "/Users/tkovs/Code/tracking/node_modules/prop-types/checkPropTypes.js",
                    "lineNumber": 82,
                    "methodName": "checkPropTypes",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 5,
                    "file": "/Users/tkovs/Code/tracking/node_modules/react/cjs/react.development.js",
                    "lineNumber": 1796,
                    "methodName": "validatePropTypes",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 5,
                    "file": "/Users/tkovs/Code/tracking/node_modules/react/cjs/react.development.js",
                    "lineNumber": 1969,
                    "methodName": "Object.createElementWithValidation [as createElement]",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 36,
                    "file": "/Users/tkovs/Code/tracking/src/screens/PacketInfo/Header/__tests__/Header.test.js",
                    "lineNumber": 9,
                    "methodName": "Object.<anonymous>",
                  },
                ],
                "symbolicated": Object {
                  "error": null,
                  "stack": null,
                  "status": "NONE",
                },
              },
            ],
          },
        ]
      }
      disableVirtualization={false}
      horizontal={false}
      initialNumToRender={10}
      maxToRenderPerBatch={10}
      numColumns={1}
      onEndReachedThreshold={2}
      removeClippedSubviews={false}
      scrollEnabled={false}
      scrollEventThrottle={50}
      scrollsToTop={false}
      stickyHeaderIndices={Array []}
      style={
        Object {
          "height": 48.5,
        }
      }
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View
        style={null}
      >
        <TouchableWithoutFeedback>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "rgba(250, 186, 48, 0.95)",
                },
                Object {
                  "height": 48,
                  "justifyContent": "center",
                  "marginTop": 0.5,
                  "paddingHorizontal": 12,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "alignItems": "flex-start",
                  "flexDirection": "row",
                }
              }
            >
              <Text
                numberOfLines={2}
                style={
                  Object {
                    "color": "rgba(255, 255, 255, 1)",
                    "flex": 1,
                    "fontSize": 14,
                    "includeFontPadding": false,
                    "lineHeight": 18,
                  }
                }
              >
                <Text>
                  Warning: Failed prop type: The prop \`title\` is marked as required in \`Header\`, but its value is \`undefined\`.
    in Header
                </Text>
              </Text>
            </View>
          </View>
        </TouchableWithoutFeedback>
      </View>
    </ScrollView>
    <SafeAreaView
      style={
        Object {
          "backgroundColor": "rgba(250, 186, 48, 0.95)",
          "marginTop": 0.5,
        }
      }
    />
  </View>
</View>
`;
